packs/influxdb/templates/influxdb.nomad.tpl

```diff
@@ -0,0 +1,164 @@
+job [[ template "job_name" . ]] {
+  [[ template "region" . ]]
+  datacenters = [[ .influxdb.datacenters | toPrettyJson ]]
+  type = "service"
+  [[- if .influxdb.namespace ]]
+  namespace   = [[ .influxdb.namespace | quote ]]
+  [[- end]]
+  [[- if .influxdb.constraints ]][[ range $idx, $constraint := .influxdb.constraints ]]
+  constraint {
+    [[- if ne $constraint.attribute "" ]]
+    attribute = [[ $constraint.attribute | quote ]]
+    [[- end ]]
+    [[- if ne $constraint.value "" ]]
+    value     = [[ $constraint.value | quote ]]
+    [[- end ]]
+    [[- if ne $constraint.operator "" ]]
+    operator  = [[ $constraint.operator | quote ]]
+    [[- end ]]
+  }
+  [[- end ]]
+  [[- end ]]
+
+  group [[ template "job_name" . ]] {
+    count = 1
+
+    network {
+      port "http" {
+        to = 8086
+      }
+    }
+
+    [[- if .influxdb.register_consul_service ]]
+    service {
+      name = "[[ .influxdb.consul_service_name ]]"
+      [[if ne (len .influxdb.consul_service_tags) 0 ]]
+      tags = [ [[ range $idx, $tag := .influxdb.consul_service_tags ]][[if $idx]],[[end]][[ $tag | quote ]][[ end ]] ]
+      [[ end ]]
+      port = "http"
+
+      check {
+        name     = "alive"
+        type     = "http"
+        path     = "/health"
+        interval = "10s"
+        timeout  = "2s"
+      }
+    }
+    [[- end ]]
+
+    [[- if .influxdb.config_volume_name ]]
+    volume "[[.influxdb.config_volume_name]]" {
+      type      = "[[.influxdb.config_volume_type]]"
+      read_only = false
+      source    = "[[.influxdb.config_volume_name]]"
+    }
+    [[- end]]
+
+    [[- if .influxdb.data_volume_name ]]
+    volume "[[.influxdb.data_volume_name]]" {
+      type      = "[[.influxdb.data_volume_type]]"
+      read_only = false
+      source    = "[[.influxdb.data_volume_name]]"
+    }
+    [[- end]]
+
+    restart {
+      attempts = 2
+      interval = "30m"
+      delay = "15s"
+      mode = "fail"
+    }
+
+    [[- if .influxdb.data_volume_name ]]
+    task "chown" {
+        lifecycle {
+            hook = "prestart"
+            sidecar = false
+        }
+
+        volume_mount {
+          volume      = "[[ .influxdb.data_volume_name ]]"
+          destination = "/var/lib/influxdb2"
+          read_only   = false
+        }
+
+        driver = "docker"
+        config {
+          image   = "busybox:stable"
+          command = "sh"
+          args    = ["-c", "chown -R 1000:1000 /var/lib/influxdb2"]
+        }
+
+        resources {
+          cpu    = 200
+          memory = 128
+        }
+    }
+    [[- end]]
+
+    [[- if .influxdb.config_volume_name ]]
```

Comment:
If I run this with the `vars.nomad` file I get the following error in the Nomad logs.

```shell
	* Task group influxdb validation failed: 1 error occurred:
	* Task 2 redefines 'chown' from task 1
```

I think you need to name these two tasks with distinct names. They can't both be named `chown` if it is allowed that both `config_volume_name` and `data_volume_name` can be set.

Reply:
Thanks for point it out! Fix is on the way.